wandb/data_types.py

```diff
@@ -522,6 +528,76 @@ def captions(audio_list):
             return ['' if c == None else c for c in captions]
 
 
+def isNumpyArray(data):
+    np = util.get_module(
+        "numpy", required="Logging Raw Point cloud data requires numpy")
+    return isinstance(data, np.ndarray)
+
+
+class Object3D(IterableMedia):
+    MAX_3D_COUNT = 20
+    SUPPORTED_TYPES = set(['obj', 'gltf', 'babylon', 'stl'])
+
+    def __init__(self, data, **kwargs):
+        """
+        Accepts a 3D File or numpy array
+        """
+        if hasattr(data, 'read'):
+            if hasattr(data, 'seek'):
+                data.seek(0)
+            self.object3D = data.read()
+            try:
+                extension = os.path.splitext(data.name)[1][1:]
+            except:
+                raise ValueError(
+                    "Object3D must be called with a python file object that has a name attribute")
+            if extension in Object3D.SUPPORTED_TYPES:
+                self.extension = extension
+            else:
+                raise ValueError("Object 3D only supports numpy arrays or files of the type: " +
+                                 ", ".join(Object3D.SUPPORTED_TYPES))
+        elif isNumpyArray(data):
+            self.numpyData = data
+        else:
+            raise ValueError("data must be a numpy or a file object")
+
+    @staticmethod
+    def transform(threeD_list, out_dir, key, step):
+        if len(threeD_list) > Object3D.MAX_3D_COUNT:
+            logging.warn(
+                "The maximum number of Object3D files to store per key is %i." % Object3D.MAX_3D_COUNT)
+        base_path = os.path.join(out_dir, "media", "object3D")
+        util.mkdir_exists_ok(base_path)
+        truncated = threeD_list[:Object3D.MAX_3D_COUNT]
+
+        filenames = []
+
+        for i, obj in enumerate(truncated):
+            # Encode the numpy array as json and send it to the server so we can use it
+            # later when needed.
+            if hasattr(obj, "numpyData"):
+                data = obj.numpyData.tolist()
+                filename = "point_cloud_key:{}_step:{}_i:{}.pts.json".format(
```

Comment:
Feels funky to have this complex file name in the raw data case, and a simple filename with just underscores in the formatted case.  Also, would it make sense to convert raw numpyData into one of the supported formats?  I would think we should atleast verify the dimensions of the numpy data.